src/AzSK.ADO/Framework/Abstracts/MetaInfoProvider.ps1

```diff
@@ -0,0 +1,284 @@
+        if ($null -ne $adoInfoAPI -and $Env:AzSKADODoNotUseADOInfoAPI -ne $true)
+        {
+            #TODO
+            #$adoInfoAPI.Enabled = $false;
+            if ($adoInfoAPI.Enabled)
+            {
+                $this.bUseADOInfoAPI = $true;
+                $this.code = $adoInfoAPI.Code;
+                $this.baseURL = $adoInfoAPI.Endpoint;
+            }
+		}
+		return $this.bUseADOInfoAPI;
+	}
+
+	[PSObject] CallADOInfoAPI($queryString )
+	{
+        $adoInfoInvokeURL = $this.baseURL + $this.FuncAPI + $queryString
+        $Header = @{
+            "x-functions-key" = $this.code;
+        }
+        $rsrcList = $null
+        
+        try 
+        {
+            $rsrcList = Invoke-RestMethod -Method 'GET' -Uri $adoInfoInvokeURL -Headers $Header
+        }
+        catch
+        {
+            Write-Host "Error calling ADO Info API. `r`nPlease contact your project's ADO security team." -ForegroundColor Red 
+        }
+        return $rsrcList;
+	}
+
+	[void] FetchMappingFiles($ResourceTypeName)
+	{
+		if ($ResourceTypeName -in ([ResourceTypeName]::Build, [ResourceTypeName]::All, [ResourceTypeName]::Build_Release, [ResourceTypeName]::Build_Release_SvcConn_AgentPool_User))
+		{
+		   if (!$this.buildSTDetails) {
+                $this.buildSTDetails = [ConfigurationManager]::LoadServerConfigFile("BuildSTData.json");
+               }	
+		}
+
+		if ($ResourceTypeName -in ([ResourceTypeName]::Release, [ResourceTypeName]::All, [ResourceTypeName]::Build_Release, [ResourceTypeName]::Build_Release_SvcConn_AgentPool_User))
+		{
+			if (!$this.releaseSTDetails) {
+				$this.releaseSTDetails = [ConfigurationManager]::LoadServerConfigFile("ReleaseSTData.json");
+			}
+		}
+
+		if ($ResourceTypeName -in ([ResourceTypeName]::ServiceConnection, [ResourceTypeName]::All, [ResourceTypeName]::Build_Release_SvcConn_AgentPool_User))
+		{
+			if (!$this.svcConnSTDetails) {
+				$this.svcConnSTDetails = [ConfigurationManager]::LoadServerConfigFile("ServiceConnectionSTData.json");
+			}
+		}
+		if ($ResourceTypeName -in ([ResourceTypeName]::AgentPool, [ResourceTypeName]::All, [ResourceTypeName]::Build_Release_SvcConn_AgentPool_User))
+		{
+			if (!$this.agtPoolSTDetails) {
+				$this.agtPoolSTDetails = [ConfigurationManager]::LoadServerConfigFile("AgentPoolSTData.json");
+			}
+		}
+
+		if ($ResourceTypeName -in ([ResourceTypeName]::VariableGroup, [ResourceTypeName]::All))
+		{
+			if (!$this.varGroupSTDetails) {
+				$this.varGroupSTDetails = [ConfigurationManager]::LoadServerConfigFile("VariableGroupSTData.json");
+			}
+        }
+        if ($ResourceTypeName -eq "ServiceTree")
+		{
+			if (!$this.serviceTreeDetails) {
+				$this.serviceTreeDetails = [ConfigurationManager]::LoadServerConfigFile("ServiceTreeData.json");
+			}
+		}
+    }
+    
+    [PSObject] FetchServiceAssociatedResources($svcId, $projectName, $ResourceTypeName)
+    {
+        $rsrcList = $null;
+        if ($this.bUseADOInfoAPI -eq $true -or $this.CheckIsADOInfoAPIEnabled())
```

Comment:
first condition is not reqd as the variable will not be set first time.

Reply:
removed added, second condition added in constructed to call once,